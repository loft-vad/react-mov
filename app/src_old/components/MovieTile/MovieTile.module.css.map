{"version":3,"sourceRoot":"","sources":["MovieTile.module.scss"],"names":[],"mappings":"AACE;EACE;EAQA;;AANA;EACE;EACA;EACA;;AAMJ;EACE;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;;AAGF;EACE;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;;AAGF;EACE;EACA;EACA;EACA;EACA;EACA;EACA;EACA","file":"MovieTile.module.css"}