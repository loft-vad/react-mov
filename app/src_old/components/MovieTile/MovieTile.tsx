import React from "react";
import styles from "./MovieTile.module.scss";
import { Link } from "react-router-dom";

export interface Movie {
  id: number;
  poster_path: string;
  title: string;
  releaseYear: number;
  genres: string[];
}

interface MovieTileProps {
  movieData: Movie;
  onClickHandler?: () => void;
}

const MovieTile: React.FC<MovieTileProps> = ({
  movieData: { poster_path, title, releaseYear, genres, id },
  onClickHandler,
}) => {
  return (
    <div className={styles.movieTileWrapper} onClick={onClickHandler}>
      <Link to={"" + id}>
        <div className={styles.movieImage}>
          <img src={poster_path} alt={title} />
        </div>
        <div className={styles.movieTitle}>
          {title}
          <div className={styles.moviewYear}>{releaseYear}</div>
        </div>
        <div className={styles.movieGenres}>
          {genres.map((genre, index) => (
            <span key={genre + index}>{genre} </span>
          ))}
        </div>
      </Link>
    </div>
  );
};

export default MovieTile;
