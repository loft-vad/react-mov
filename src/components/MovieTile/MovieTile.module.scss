.movieTileWrapper {
  .movieImage {
    width: 323px;

    img {
      display: block;
      width: 100%;
      aspect-ratio: 9 / 13;
    }

    padding-bottom: 20px;
  }

  .movieTitle {
    padding-bottom: 5px;
    font-weight: 500;
    font-size: 18px;
    line-height: 22px;
    color: #FFFFFF;
    mix-blend-mode: normal;
    opacity: 0.7;
    display: flex;
    align-items: center;
    justify-content: space-between;
  }

  .moviewYear {
    font-family: 'Montserrat';
    font-style: normal;
    font-weight: 500;
    font-size: 14px;
    line-height: 17px;
    color: #FFFFFF;
    mix-blend-mode: normal;
    opacity: 0.7;
    padding: 10px 15px;
    border: 1px solid #555555;
    border-radius: 4px;
    flex: 0 0 auto;
  }

  .movieGenres {
    font-family: 'Montserrat';
    font-style: normal;
    font-weight: 500;
    font-size: 14px;
    line-height: 17px;
    color: #FFFFFF;
    mix-blend-mode: normal;
    opacity: 0.5;
  }
}