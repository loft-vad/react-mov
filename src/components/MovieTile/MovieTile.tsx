import React from "react";
import styles from "./MovieTile.module.scss";

export interface Movie {
  id: number;
  imageUrl: string;
  movieName: string;
  releaseYear: number;
  genres: string[];
}

interface MovieTileProps {
  movieData: Movie;
  onClickHandler: () => void;
}

const MovieTile: React.FC<MovieTileProps> = ({
  movieData: { imageUrl, movieName, releaseYear, genres },
  onClickHandler,
}) => {
  return (
    <div className={styles.movieTileWrapper} onClick={onClickHandler}>
      <div className={styles.movieImage}>
        <img src={imageUrl} alt={movieName} />
      </div>
      <div className={styles.movieTitle}>
        {movieName}
        <div className={styles.moviewYear}>{releaseYear}</div>
      </div>
      <div className={styles.movieGenres}>
        {genres.map((genre, index) => (
          <span key={genre + index}>{genre} </span>
        ))}
      </div>
    </div>
  );
};

export default MovieTile;
