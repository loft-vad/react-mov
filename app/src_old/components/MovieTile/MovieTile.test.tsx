import React from "react";

import MovieTile, { Movie } from "./MovieTile";
import { fireEvent, render } from "@testing-library/react";

const movie: Movie = {
  id: 1,
  title: "Pulp Fiction",
  poster_path: "",
  releaseYear: 2004,
  genres: ["Action & Adventure", "Comedy"],
};

describe("MovieTile component", () => {
  it("should match snapshot", () => {
    const wrapper = render(<MovieTile movieData={movie} onClickHandler={() => {}} />);

    expect(wrapper.container.firstChild).toMatchSnapshot();
  });

  it("after clicking on movie tile onClickHandler is called", () => {
    const onClickHandler = jest.fn();
    const wrapper = render(<MovieTile movieData={movie} onClickHandler={onClickHandler} />);
    const movieTileWrapper = wrapper.container.querySelector("div");

    expect(movieTileWrapper).toBeInTheDocument();
    fireEvent.click(movieTileWrapper!);
    expect(onClickHandler).toHaveBeenCalled();
  });
});
