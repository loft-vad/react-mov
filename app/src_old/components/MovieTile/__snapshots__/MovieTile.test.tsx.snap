// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieTile component should match snapshot 1`] = `
<div
  class="movieTileWrapper"
>
  <div
    class="movieImage"
  >
    <img
      alt="Pulp Fiction"
      src=""
    />
  </div>
  <div
    class="movieTitle"
  >
    Pulp Fiction
    <div
      class="moviewYear"
    >
      2004
    </div>
  </div>
  <div
    class="movieGenres"
  >
    <span>
      Action & Adventure
       
    </span>
    <span>
      Comedy
       
    </span>
  </div>
</div>
`;
